chromium_src/chrome/browser/ui/webui/tab_search/tab_search_page_handler.cc

```diff
@@ -0,0 +1,255 @@
+    base::expected<std::vector<std::string>, ai_chat::mojom::APIError> result) {
+  if (!result.has_value()) {
+    std::move(callback).Run(false, GetError(result.error()));
+    return;
+  }
+
+  // Move all tabs from normal browser window to a new window.
+  std::vector<TabDetails> tab_details_before_move;
+  for (const auto& tab_id_str : *result) {
+    int tab_id = 0;
+    if (!base::StringToInt(tab_id_str, &tab_id)) {
+      continue;
+    }
+
+    std::optional<TabDetails> details = GetTabDetails(tab_id);
+    if (!details) {
+      continue;
+    }
+
+    // Store all details before move to preserve the index.
+    tab_details_before_move.push_back(*details);
+    // Store old window ID (session ID), tab ID, and tab strip index for
+    // undo.
+    focus_tabs_info_[details->browser->session_id()].push_back(
+        {tab_id, details->GetIndex()});
+  }
+
+  if (tab_details_before_move.empty()) {
+    std::move(callback).Run(false, nullptr);
+    return;
+  }
+
+  Browser* new_browser =
+      Browser::Create(Browser::CreateParams(Profile::FromWebUI(web_ui_), true));
+  for (const auto& details : tab_details_before_move) {
+    std::unique_ptr<tabs::TabModel> tab =
+        details.browser->tab_strip_model()->DetachTabAtForInsertion(
+            details.GetIndex());
+    new_browser->tab_strip_model()->AppendTab(std::move(tab),
+                                              false /* foreground */);
+  }
+  new_browser->window()->Show();
+
+  std::move(callback).Run(true, nullptr);
+}
+
+void TabSearchPageHandler::UndoFocusTabs(UndoFocusTabsCallback callback) {
+  for (auto& iter : focus_tabs_info_) {
+    // Find the browser with the session ID (key).
+    Browser* target = nullptr;
+    for (Browser* browser : *BrowserList::GetInstance()) {
+      if (!ShouldTrackBrowser(browser)) {
+        continue;
+      }
+
+      if (browser->session_id() == iter.first) {
+        target = browser;
+        break;
+      }
+    }
+
+    if (!target) {
+      continue;
+    }
+
+    // Sort the tabs info by index in ascending order to ensure tabs are
+    // inserted in the correct order.
+    std::ranges::sort(iter.second, [](const TabInfo& a, const TabInfo& b) {
+      return a.index < b.index;
+    });
+
+    for (const auto& tab_info : iter.second) {
+      // Find the moved tab.
+      std::optional<TabDetails> details = GetTabDetails(tab_info.tab_id);
+      if (!details) {
+        continue;
+      }
+
+      std::unique_ptr<tabs::TabModel> tab =
+          details->browser->tab_strip_model()->DetachTabAtForInsertion(
+              details->GetIndex());
+      target->tab_strip_model()->InsertDetachedTabAt(
+          tab_info.index, std::move(tab), AddTabTypes::ADD_NONE);
+    }
+  }
+
+  focus_tabs_info_.clear();
+  std::move(callback).Run();
+}
+
+void TabSearchPageHandler::OpenLeoGoPremiumPage() {
+  NavigateParams params(
+      Profile::FromWebUI(web_ui_),
+      GURL("https://account.brave.com/account/?intent=checkout&product=leo"),
```

Comment:
Do we have a constant with this URL in it somewhere else?

Reply:
Only one in an anonymous namespace in ai_chat_ui_page_handler.cc.

Reply:
Hmm, I wonder if we should move it to `constants.h` in the AIChat component?

Reply:
sure, will do

Reply:
947c6ac1eb1cee23d4622a3450e6f0339d751d0e